sdk/search/azure-search-documents/src/test/java/com/azure/search/documents/FieldBuilderTest.java

```diff
@@ -0,0 +1,138 @@
+        Field description = new SimpleField("description", DataType.EDM_STRING, false).build();
+        Field category = new SimpleField("category", DataType.EDM_STRING, false).build();
+        Field tags = new SearchableField("tags", true).build();
+        Field parkingIncluded = new SimpleField("parkingIncluded", DataType.EDM_BOOLEAN, false).build();
+        Field smokingAllowed = new SimpleField("smokingAllowed", DataType.EDM_BOOLEAN, false).build();
+        Field lastRenovationDate = new SimpleField("lastRenovationDate", DataType.EDM_DATE_TIME_OFFSET, false).build();
+        Field rating = new SimpleField("rating", DataType.EDM_INT32, false).build();
+        Field location = new SimpleField("location", DataType.EDM_GEOGRAPHY_POINT, false).build();
+        Field address = new ComplexField("address", false)
+            .setFields(buildHotelAddressField()).build();
+        Field rooms = new ComplexField("rooms", true).setFields(buildHotelRoomField()).build();
+
+        return Arrays.asList(hotelId, hotelName, description, category, tags, parkingIncluded, smokingAllowed,
+            lastRenovationDate, rating, location, address, rooms);
+    }
+
+    private List<Field> buildHotelAddressField() {
+        Field streetAddress = new SimpleField("streetAddress", DataType.EDM_STRING, false).setFacetable(true)
+            .setKey(true).build();
+        Field city = new SearchableField("city", false).setFilterable(true).build();
+        Field stateProvince = new SearchableField("stateProvince", false).build();
+        Field country = new SearchableField("country", false)
+            .setSynonymMaps(Arrays.asList("America -> USA", "USA -> US")).build();
+        Field postalCode = new SimpleField("postalCode", DataType.EDM_STRING, false).build();
+        return Arrays.asList(streetAddress, city, stateProvince, country, postalCode);
+    }
+
+    private List<Field> buildHotelRoomField() {
+        Field description = new SimpleField("description", DataType.EDM_STRING, false).build();
+        Field descriptionFr = new SimpleField("descriptionFr", DataType.EDM_STRING, false).build();
+        Field type = new SimpleField("type", DataType.EDM_STRING, false).build();
+        Field baseRate = new SimpleField("baseRate", DataType.EDM_DOUBLE, false).build();
+        Field bedOptions = new SimpleField("bedOptions", DataType.EDM_STRING, false).build();
+        Field sleepsCount = new SimpleField("sleepsCount", DataType.EDM_INT32, false).build();
+        Field smokingAllowed = new SimpleField("smokingAllowed", DataType.EDM_BOOLEAN, false).build();
+        Field tags = new SimpleField("tags", DataType.EDM_STRING, true).build();
+        return Arrays.asList(description, descriptionFr, type, baseRate, bedOptions, sleepsCount, smokingAllowed, tags);
+    }
+
+    private List<Field> sortByFieldName(List<Field> fields) {
+        return fields.stream().sorted((o1, o2) -> o1.getName().compareTo(o2.getName())).collect(Collectors.toList());
```

Comment:
Have you tried `Collections.sort(fields, (o1, o2) -> {})`? It'll save you from having to turn this from a list to a stream, then back into a list.